widget/card.go

```diff
@@ -0,0 +1,204 @@
+package widget
+
+import (
+	"image/color"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/internal/widget"
+	"fyne.io/fyne/theme"
+)
+
+type cardRenderer struct {
+	*widget.ShadowingRenderer
+
+	header, subHeader *canvas.Text
+
+	card *CardContainer
+}
+
+const (
+	cardMediaHeight = 128
+)
+
+// MinSize calculates the minimum size of a card.
+// This is based on the contained text, image and content.
+func (c *cardRenderer) MinSize() fyne.Size {
+	hasHeader := c.card.Title != ""
+	hasSubHeader := c.card.SubTitle != ""
+	hasImage := c.card.Image != nil
+	hasContent := c.card.Content != nil
+
+	if !hasHeader && !hasSubHeader && !hasContent {
+		return fyne.NewSize(c.card.Image.MinSize().Width+theme.Padding(), cardMediaHeight+theme.Padding())
+	}
+
+	min := fyne.NewSize(theme.Padding()*5, theme.Padding()*5) // content padding plus 1 pad border
```

Comment:
Maybe this is a convention used elsewhere in the code that I haven't experienced, but I see a lot of 5s and 3s that seem a little arbitrary? Maybe those should be non-exported `const` variables so they can be documented?

Reply:
Hmm, I would often agree but in this case it is somewhat arbitrary... This was tweaking layouts to align things nicely for the component. 
I could write it out 5 as borderPadding + horizontalContentPadding*2 having defined a few new variables - is it worth it?

Reply:
I think it's worth breaking this kind of stuff out at least into consts with comments, since it makes it easier for a future reader to see what you meant / where you got the numbers from. Even if it's just an arbitrary magic number, explaining how you arrived at that value might be useful to future maintainers.

Reply:
should be fixed now